src/features/comments/comments.ts

```diff
@@ -0,0 +1,91 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Comment } from '../../types/Comment';
+import {
+  createComment,
+  deleteComment,
+  getPostComments,
+} from '../../api/comments';
+
+type InitialState = {
+  comments: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: InitialState = {
+  comments: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const init = createAsyncThunk('comments/fetch', (postId: number) => {
+  return getPostComments(postId);
+});
+
+export const deleteComments = createAsyncThunk(
+  'comments/delete',
+  async (commentId: number) => {
+    await deleteComment(commentId);
+
+    return commentId;
+  },
+);
+
+export const addComments = createAsyncThunk(
+  'comments/post',
+  (comment: Comment) => {
+    const newComment = createComment(comment);
+
+    return newComment;
+  },
+);
+
+const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {},
+  extraReducers(builder) {
+    builder.addCase(init.pending, state => {
+      return { ...state, loaded: false };
+    });
+
+    builder.addCase(
+      init.fulfilled,
+      (state, action: PayloadAction<Comment[]>) => {
+        return { ...state, loaded: true, comments: action.payload };
+      },
+    );
+
+    builder.addCase(init.rejected, state => {
+      return { ...state, hasError: true };
```

Comment:
In the `init.rejected` case, consider setting `loaded` to `true` to indicate that the loading process has completed, even though it resulted in an error.